servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -529,30 +529,22 @@ void convertToLowerCase(Object obj) {
             @Override
             void convertToLowerCase(Object obj) {
                 AssertionConditions assertionConditions = (AssertionConditions) obj;
-                // lower case key and value, assertion condition operator is an ENUM, hence left unchanged
-                assertionConditions.setConditionsList(assertionConditions.getConditionsList().stream()
-                        .map(c -> new AssertionCondition()
-                                .setConditionsMap(c.getConditionsMap().entrySet().stream()
-                                        .collect(Collectors.toMap(
-                                                e -> e.getKey().toLowerCase(),
-                                                e -> e.getValue().setValue(e.getValue().getValue().toLowerCase())
-                                        ))))
-                        .collect(Collectors.toList())
-                );
+                    for (AssertionCondition assertionCondition : assertionConditions.getConditionsList()) {
+                        ASSERTION_CONDITION.convertToLowerCase(assertionCondition);
+                    }
             }
         },
         ASSERTION_CONDITION {
             @Override
             void convertToLowerCase(Object obj) {
                 AssertionCondition assertionCondition = (AssertionCondition) obj;
                 // lower case key and value, assertion condition operator is an ENUM, hence left unchanged
-                if (assertionCondition.getConditionsMap() != null) {
-                    assertionCondition.setConditionsMap(assertionCondition.getConditionsMap().entrySet().stream()
-                            .collect(Collectors.toMap(
-                                    e -> e.getKey().toLowerCase(),
-                                    e -> e.getValue().setValue(e.getValue().getValue().toLowerCase())
-                            )));
+                Map<String, AssertionConditionData> lowerCasedMap = new HashMap<>();
```

Comment:
Here we're still creating a new map set. My original comment was whether or not we can implement this without creating anything new. If that's not possible, then the original implementation using streams is preferred. also, the original code correctly checked for null for conditions map while this version does not.

Reply:
validator should take care of null since the individual properties in AssertionCondition are mandatory.